flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -159,128 +156,180 @@ public String getJobName() {
-		this.allVerticesInSameSlotSharingGroupByDefault = allVerticesInSameSlotSharingGroupByDefault;
+	public JobGraph getJobGraph() {
+		return getJobGraph(null);
 	}
 
 	/**
-	 * Gets whether to put all vertices into the same slot sharing group by default.
-	 *
-	 * @return whether to put all vertices into the same slot sharing group by default.
+	 * Gets the assembled {@link JobGraph} with a specified {@link JobID}.
 	 */
-	public boolean isAllVerticesInSameSlotSharingGroupByDefault() {
-		return allVerticesInSameSlotSharingGroupByDefault;
+	public JobGraph getJobGraph(@Nullable JobID jobID) {
+		return StreamingJobGraphGenerator.createJobGraph(this, jobID);
 	}
 
-	// Checkpointing
+	public String getStreamingPlanAsJSON() {
+		try {
+			return new JSONGenerator(this).getJSON();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("JSON plan creation failed", e);
+		}
+	}
 
-	public boolean isChainingEnabled() {
-		return chaining;
+	//-----------------------------------------------------------------------------------------
+	//  MUTATOR METHODS - used exclusively from the StreamGraphGenerator
+	//-----------------------------------------------------------------------------------------
+
+	void setJobName(String jobName) {
+		this.jobName = jobName;
 	}
 
-	public boolean isIterative() {
-		return !vertexIDtoLoopTimeout.isEmpty();
+	void setChaining(boolean chaining) {
+		this.chaining = chaining;
+	}
+
+	void setStateBackend(StateBackend backend) {
+		this.stateBackend = backend;
+	}
+
+	void setScheduleMode(ScheduleMode scheduleMode) {
+		this.scheduleMode = scheduleMode;
+	}
+
+	void setUserArtifacts(Collection<Tuple2<String, DistributedCache.DistributedCacheEntry>> userArtifacts) {
+		this.userArtifacts = userArtifacts;
+	}
+
+	void setTimeCharacteristic(TimeCharacteristic timeCharacteristic) {
+		this.timeCharacteristic = timeCharacteristic;
+	}
+
+	/**
+	 * Set whether to put all vertices into the same slot sharing group by default.
+	 *
+	 * @param allVerticesInSameSlotSharingGroupByDefault indicates whether to put all vertices
+	 *                                                   into the same slot sharing group by default.
+	 */
+	void setAllVerticesInSameSlotSharingGroupByDefault(boolean allVerticesInSameSlotSharingGroupByDefault) {
+		this.allVerticesInSameSlotSharingGroupByDefault = allVerticesInSameSlotSharingGroupByDefault;
 	}
 
-	public <IN, OUT> void addSource(
+	<OUT> void addSource(
 			Integer vertexID,
 			@Nullable String slotSharingGroup,
 			@Nullable String coLocationGroup,
 			SourceOperatorFactory<OUT> operatorFactory,
-			TypeInformation<IN> inTypeInfo,
 			TypeInformation<OUT> outTypeInfo,
 			String operatorName) {
 		addOperator(
-				vertexID,
-				slotSharingGroup,
-				coLocationGroup,
-				operatorFactory,
-				inTypeInfo,
-				outTypeInfo,
-				operatorName,
-				SourceOperatorStreamTask.class);
+			vertexID,
```

Comment:
I think it is worth keeping the extra tab here (as before) to separate args from body of the method with indentation.